admin/Openedx_Woocommerce_Plugin_Admin.php

```diff
@@ -105,85 +110,138 @@ public function enqueue_styles() {
+	{
+		global $post;
+
+		echo '<div class="options_group">';
+
+		echo '<p class="form-field">' . __('Only use these fields if the product is an Open edX course.', 'woocommerce') . '</p>';
+
+		woocommerce_wp_text_input(array(
+			'id'          => '_course_id',
+			'label'       => __('Course ID', 'woocommerce'),
+			'placeholder' => '',
+			'desc_tip'    => 'true',
+			'description' => __('Enter the course ID for the product.', 'woocommerce')
+		));
+
+		woocommerce_wp_select(array(
+			'id'          => '_mode',
+			'label'       => __('Mode', 'woocommerce'),
+			'desc_tip'    => 'true',
+			'description' => __('Select the mode for the product.', 'woocommerce'),
+			'options'     => array(
+				'Honor'             => __('Honor', 'woocommerce'),
+				'Audit'             => __('Audit', 'woocommerce'),
+				'Verified'          => __('Verified', 'woocommerce'),
+				'Credit'            => __('Credit', 'woocommerce'),
+				'Professional'      => __('Professional', 'woocommerce'),
+				'No ID Professional' => __('No ID Professional', 'woocommerce'),
+			)
+		));
+
+		echo '</div>';
 	}
 
-	 /**
-     * Wrapper function to register a new post type
-     *
-     * @param  string $post_type   Post type name.
-     * @param  string $plural      Post type item plural name.
-     * @param  string $single      Post type item single name.
-     * @param  string $description Description of post type.
-     * @return object              Post type class object
-     */
-    public function register_post_type( $post_type = '', $plural = '', $single = '', $description = '', $options = array() ) {
-
-        if ( ! $post_type || ! $plural || ! $single ) {
-            return;
-        }
-
-        $post_type = $this->createPostType($post_type, $plural, $single, $description, $options);
-
-        return $post_type;
-
-    }
-
-     /**
-     * Create a new instance of the Openedx_Woocommerce_Plugin_Post_Type class and register a new post type.
-     *
-     * @param  string $post_type   Post type name.
-     * @param  string $plural      Post type item plural name.
-     * @param  string $single      Post type item single name.
-     * @param  string $description Description of the post type.
-     * @param  array  $options     Additional options for the post type.
-     * @return object              Post type class object.
-     */
-     public function createPostType($post_type = '', $plural = '', $single = '', $description = '', $options = array()) {
-
-        return new Openedx_Woocommerce_Plugin_Post_Type($post_type, $plural, $single, $description, $options);
-
-    }
+	/**
+	 * Save course ID and mode fields for product
+	 *
+	 * @since    1.1.1
+	 */
+	function save_custom_product_fields($post_id)
+	{
+		$course_id = isset($_POST['_course_id']) ? sanitize_text_field($_POST['_course_id']) : '';
+		$mode = isset($_POST['_mode']) ? sanitize_text_field($_POST['_mode']) : '';
+
+		update_post_meta($post_id, '_course_id', $course_id);
```

Comment:
Only for curiosity, where do you store that information in the fields? That $post_id is for the product?

Reply:
Yes, the $post_id is an ID that most objects, pages, and others within Wordpress have. So, if we want to save any type of metadata in these, we will have to use this variable to tell WordPress to whom or what it should assign this property.

Reply:
Thanks